crates/neon-runtime/src/win_delay_load_hook.cc

```diff
@@ -0,0 +1,31 @@
+/*
+ * When this file is linked to a DLL, it sets up a delay-load hook that
+ * intervenes when the DLL is trying to load 'node.exe' or 'iojs.exe'
+ * dynamically. Instead of trying to locate the .exe file it'll just return
+ * a handle to the process image.
+ *
+ * This allows compiled addons to work when node.exe or iojs.exe is renamed.
+ */
+
+ 
+#include <windows.h>
+
+#include <delayimp.h>
+#include <string.h>
+
+extern "C" FARPROC WINAPI __load_exe_hook(unsigned int, DelayLoadInfo*);
```

Comment:
You don't need to export the symbol, only `__pfnDliNotifyHook2` needs to be exported. Just this should suffice:
```c
static FARPROC WINAPI __load_exe_hook(unsigned int event, DelayLoadInfo* info) {
  // ...
}

decltype(__pfnDliNotifyHook2) __pfnDliNotifyHook2 = __load_exe_hook;
```
By the way, I think you could turn this into a .rs file. You'd probably have to provide definitions of `dliNotePreLoadLibrary` and `DelayLoadInfo` though, so it might be not be shorter.

Reply:
Much better, thanks!

I have no idea how to write those definitions they're pretty complicated to me. And as neon-runtime already includes c code anyways, i figured a little more shouldn't be a problem. Though of course pure rust is nicer.

Optionally removing it is as simple as removing `.cargo/config` (though the hook then still gets needlessly compiled in neon-runtime) as long as it's documented it's there i think it shouldn't be an issue.